### PR TITLE
v.colors: Implement rules from stdin

### PR DESCRIPTION
**Pull request by HuidaeCho**
Make #1297 more efficient by supporting rules from stdin.

### CODE REVIEW

File: vector/v.colors/v.colors.html [hunk 1]
```diff
@@ -86,6 +86,17 @@ <h3>Convert RGB attribute values into color table</h3>
 with GRASSRGB attribute column. Also note that color table is preferred
 over RGB values stored in attribute table.
 
+<h3>Transfer raster colors to vector</h3>
+
+<div class="code"><pre>
+# create an example raster
+v.to.rast input=censusblk_swwake use=attr attribute_column=TOTAL_POP output=censusblk_swwake_total_pop
```
Comment (neteler) [inline vector/v.colors/v.colors.html:93]:
```suggestion
# create an example raster from census blocks (10m pixel resolution)
g.region vector=censusblk_swwake res=10 -ap
v.to.rast input=censusblk_swwake use=attr attribute_column=TOTAL_POP output=censusblk_swwake_total_pop
```

**Review by neteler — APPROVED** [review]
Nice, works as expected.

**Comment by cmbarton** [comment]
This is good. Fixing the raster= argument #1297 would make it even more efficient

**Comment by HuidaeCho** [comment]
> This is good. Fixing the raster= argument #1297 would make it even more efficient

I wouldn't say "fix" because it seems that that usage in #1297 was never intended. It would be a new feature.

**Comment by cmbarton** [comment]
> > This is good. Fixing the raster= argument #1297 would make it even more efficient
> 
> I wouldn't say "fix" because it seems that that usage in #1297 was never intended. It would be a new feature.

From v.colors manual. "The raster option allows user to specify a raster map from which to copy the color table" (this is the only explanation given)

How is this intended to be used?

**Comment by HuidaeCho** [comment]
> > > This is good. Fixing the raster= argument #1297 would make it even more efficient
> > 
> > 
> > I wouldn't say "fix" because it seems that that usage in #1297 was never intended. It would be a new feature.
> 
> From v.colors manual. "The raster option allows user to specify a raster map from which to copy the color table" (this is the only explanation given)
> 
> How is this intended to be used?

Yes, it's not clear. Based on the code, looks like that sentence means raster value to vector cat color mapping (not that useful?), not value to attribute (more useful and intuitive). Actually, we can have both.

**Comment by cmbarton** [comment]
> > > > This is good. Fixing the raster= argument #1297 would make it even more efficient
> > > 
> > > 
> > > I wouldn't say "fix" because it seems that that usage in #1297 was never intended. It would be a new feature.
> > 
> > 
> > From v.colors manual. "The raster option allows user to specify a raster map from which to copy the color table" (this is the only explanation given)
> > How is this intended to be used?
> 
> Yes, it's not clear. Based on the code, looks like that sentence means raster value to vector cat color mapping (not that useful?), not value to attribute (more useful and intuitive). Actually, we can have both.

I agree. That would be much more useful

**Comment by HuidaeCho** [comment]
> > > > > This is good. Fixing the raster= argument #1297 would make it even more efficient
> > > > 
> > > > 
> > > > I wouldn't say "fix" because it seems that that usage in #1297 was never intended. It would be a new feature.
> > > 
> > > 
> > > From v.colors manual. "The raster option allows user to specify a raster map from which to copy the color table" (this is the only explanation given)
> > > How is this intended to be used?
> > 
> > 
> > Yes, it's not clear. Based on the code, looks like that sentence means raster value to vector cat color mapping (not that useful?), not value to attribute (more useful and intuitive). Actually, we can have both.
> 
> I agree. That would be much more useful

Implemented in #1392.

**Pull request merged**